src/metadata.rs

```diff
@@ -76,49 +111,35 @@ fn convert_exif_date_time_to_chrono_date_time_fixed_offset(
         );
 
     match maybe_date {
-        chrono::LocalResult::Single(date) => Ok(date),
-        chrono::LocalResult::Ambiguous(_, _) => Err(MetadataError::from_str("Ambiguous date")),
-        chrono::LocalResult::None => Err(MetadataError {
-            message: "Invalid date".to_string(),
-        }),
+        chrono::LocalResult::Single(date) => Some(date),
+        chrono::LocalResult::Ambiguous(_, _) => {
+            eprintln!(
+                "Date time {} in {} was ambiguous",
+                file_name, exif_date_time
+            );
+            None
+        }
+        chrono::LocalResult::None => {
+            eprintln!("Date time {} in {} was invalid", file_name, exif_date_time);
+            None
+        }
     }
 }
 
-fn extract_date_time_exif_field(
-    exif: &Exif,
-    tag: Tag,
-) -> Result<Option<chrono::DateTime<FixedOffset>>, MetadataError> {
-    match exif.get_field(tag, In::PRIMARY) {
-        Some(field) => match field.value {
-            Value::Ascii(ref v) => match convert_exif_date_time_to_chrono_date_time_fixed_offset(
-                exif::DateTime::from_ascii(&v[0])?,
-            ) {
-                Ok(date_time) => Ok(Some(date_time)),
-                Err(err) => Err(err),
-            },
-            _ => Err(MetadataError {
-                message: "Exif date field is not ascii".to_string(),
-            }),
-        },
-        None => Ok(None),
-    }
+fn get_date_time_taken(path: &Path) -> Option<DateTime<FixedOffset>> {
+    let file_name = path.display().to_string();
+    open_file(path, &file_name)
+        .map(|file| get_exif_metadata(file, &file_name))
+        .flatten()
```

Comment:
flat map is called [and_then()](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then) in Rust. This would have been nice if we remembered from scala :sweat_smile: